v3/src/hooks/use-overlay-bounds.ts

```diff
@@ -1,36 +1,19 @@
-import { useEffect, useState } from "react"
-
 interface IProps {
   target: Element | null
   portal?: Element | null
 }
-interface IBounds {
-  left: number
-  top: number
-  width: number
-  height: number
-}
-export function useOverlayBounds({ target, portal }: IProps) {
-  const [overlayBounds, setOverlayBounds] = useState<IBounds | null>(null)
 
-  useEffect(() => {
-    // track the bounds of the target element relative to the portal element
-    const observer = target && new ResizeObserver(() => {
-      const portalBounds = portal?.getBoundingClientRect()
-      const targetBounds = target?.getBoundingClientRect()
-      if (targetBounds) {
-        setOverlayBounds({
-          left: targetBounds.x - (portalBounds?.x ?? 0),
-          top: targetBounds.y - (portalBounds?.y ?? 0),
-          width: targetBounds.width,
-          height: targetBounds.height
-        })
-      }
-    })
-    target && observer?.observe(target)
-
-    return () => observer?.disconnect()
-  }, [portal, target])
-
-  return overlayBounds
+export function useOverlayBounds({ target, portal }: IProps) {
+  const portalBounds = portal?.getBoundingClientRect()
+  const targetBounds = target?.getBoundingClientRect()
+  if (targetBounds) {
+    return {
+      left: targetBounds.x - (portalBounds?.x ?? 0),
+      top: targetBounds.y - (portalBounds?.y ?? 0),
+      width: targetBounds.width,
+      height: targetBounds.height
+    }
+  } else {
+    return {}
+  }
```

Comment:
Yes, `null` is the conventional return value to indicate that a React component should not render. You also don't need the else, so you can replace it with:
```suggestion
  }
  return null
```